pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -381,53 +307,42 @@ func (r *repository) Put(manifest *schema1.SignedManifest) error {
 		client, ok := UserClientFrom(r.ctx)
 		if !ok {
 			context.GetLogger(r.ctx).Errorf("Error creating user client to auto provision image stream: Origin user client unavailable")
-			return statusErr
+			return digest.Digest(""), statusErr
 		}
 
 		if _, err := client.ImageStreams(r.namespace).Create(&stream); err != nil {
 			context.GetLogger(r.ctx).Errorf("Error auto provisioning image stream: %s", err)
-			return statusErr
+			return digest.Digest(""), statusErr
 		}
 
 		// try to create the ISM again
 		if err := r.registryClient.ImageStreamMappings(r.namespace).Create(&ism); err != nil {
 			context.GetLogger(r.ctx).Errorf("Error creating image stream mapping: %s", err)
-			return err
-		}
-	}
-
-	// Grab each json signature and store them.
-	signatures, err := manifest.Signatures()
```

Comment:
This is not about the putting of signatures in etcd. `repositorymiddleware` newer [wrapped](https://github.com/openshift/origin/blob/master/pkg/dockerregistry/server/repositorymiddleware.go#L124) service.

This is about [putting](https://github.com/openshift/origin/blob/master/pkg/dockerregistry/server/repositorymiddleware.go#L406) signatures as separate [blobs.](https://github.com/openshift/origin/blob/master/Godeps/_workspace/src/github.com/docker/distribution/registry/storage/signaturestore.go#L113). 

You're right. I lost it. But this would be tricky since the `Signatures.Put()` is no longer available.